frontend/apps/filemanager/filemanagerutil.lua

```diff
@@ -24,13 +25,15 @@ end
 
 function filemanagerutil.abbreviate(path)
     if not path then return "" end
-    local home_dir_name = G_reader_settings:readSetting("home_dir_display_name")
-    if home_dir_name ~= nil then
+    if G_reader_settings:nilOrTrue("shorten_home_dir") then
         local home_dir = G_reader_settings:readSetting("home_dir") or filemanagerutil.getDefaultDir()
+        if path == home_dir then
+            return _("Home")
+        end
         local len = home_dir:len()
         local start = path:sub(1, len)
         if start == home_dir then
-            return home_dir_name .. path:sub(len+1)
+            return path:sub(len+2)
```

Comment:
> I need to stop commenting, I need help

Don't stop :) Add one here `-- remove leading / of sub-directory` (because I will surely wonder _why that +2 ?!_ next time I'll look at that code)

Reply:
Like I said, I'm a commentophile. It's just that this is almost verbatim saying the same thing.

```lua
-- Only do this if there's more than one thing.
if thing > 1 then
```

Reply:
Instead of a comment for a mystery number you can also use a self-explanatory variable name.

```lua
local leading_slash_len = 2
```

Depends a bit on the specifics.